docs/content/user-guide/en/cap/idempotence.md

```diff
@@ -24,126 +24,61 @@ This type of delivery guarantee can arise from your messaging system and your co
+        2. Release lease on a message
 ```
 
 and the "lease" we grabbed on the message in step (1) is associated with an appropriate timeout, then we are guaranteed that no matter how wrong things go, we will only actually remove the message from the queue (i.e. execute step (4)/(2)) if we have successfully committed our "work transaction".
 
 ### What is a "work transaction"?
 
-It depends on what you're doing 😄 maybe it's a transaction in a relational database (which traditionally have pretty good support in this regard), maybe it's a transaction in a document database that happens to support transaction (like RavenDB or Postgres), or maybe it's a conceptual transaction in the form of whichever work you happen to carry out as a consequence of handling a message, e.g. update a bunch of documents in MongoDB, move some files around in the file system, or mutate some obscure in-mem data structure.
+It depends on what you're doing 😄 maybe it's a transaction in a relational database (which traditionally has pretty good support in this regard), maybe it's a transaction in a document database that happens to support transaction (like RavenDB or Postgres), or maybe it's a conceptual transaction in the form of whichever work you happen to carry out as a consequence of handling a message, e.g. update a bunch of documents in MongoDB, move some files around in the file system, or mutate some obscure in-mem data structure.
 
 The fact that the "work transaction" is just a conceptual thing is what makes it impossible to support the aforementioned Exactly Once delivery guarantee – it's just not generally possible to commit or roll back a "work transaction" and a "queue transaction" (which is what we could call the protocol carried out with the message queue systems) atomically and consistently.
 
 ## Idempotence at CAP
 
-In CAP, **At Least Once**  delivery guarantee is used.
+In CAP,  **At Least Once**  delivery guarantee is used.
 
 Since we have a temporary storage medium (database table), we may be able to do At Most Once, but in order to strictly guarantee that the message will not be lost, we do not provide related functions or configurations.
 
-### Why are we not providing(achieving) idempotency ?
+### Why are we not providing(achieving) idempotency?
 
 1. The message was successfully written, but the execution of the Consumer method failed.  
 
     There are a lot of reasons why the Consumer method fails. I don't know if the specific scene is blindly retrying or not retrying is an incorrect choice.
-    For example, if the consumer is debiting service, if the execution of the debit is successful, but fails to write the debit log, the CAP will judge that the consumer failed to execute and try again. If the client does not guarantee idempotency, the framework will retry it, which will inevitably lead to serious consequences for multiple debits.
-
-2. The execution of the Consumer method succeeded, but received the same message.  
```

Comment:
Hi, This section and below part appears to have been removed?

Reply:
Thanks for telling me, I added that part again